File: Data Structures & Algorithms/max-consecutive-ones/submission-7.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr = 0
        streak = 0

        for i in range(len(nums)):
            if nums[i] == 1:
                curr += 1
            else:
                curr = 0
            streak = max(curr, streak)

        return streak

File: Data Structures & Algorithms/house-robber/submission-7.cpp
/*
Ok lets try a top-down DFS approach first
*/

#include <vector>
#include <algorithm> //max

// dont ever actually do this live but we're going to do this because we're speedmaxxing
using namespace std;

class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> memo(nums.size(), -1);

        return dp(nums, memo, 0);
    }

    int dp(vector<int>& nums, vector<int>& memo, int i) 
    {
        //base case - out of bounds
        if (i >= nums.size()) return 0;

        if (memo[i] != -1) return memo[i];

        memo[i] = max(dp(nums, memo, i+1), nums[i] + dp(nums, memo, i+2));
        return memo[i];
    }
};
